Common/Securities/MarketHoursDatabase.cs

```diff
@@ -95,7 +95,11 @@ public SecurityExchangeHours GetExchangeHours(SubscriptionDataConfig configurati
         /// <returns>The exchange hours for the specified security</returns>
         public SecurityExchangeHours GetExchangeHours(string market, Symbol symbol, SecurityType securityType, DateTimeZone overrideTimeZone = null)
         {
-            var stringSymbol = symbol == null ? string.Empty : symbol.Value;
+            var stringSymbol = symbol == null ? 
+                                    string.Empty :
+                                    (symbol.ID.SecurityType == SecurityType.Future? 
+                                        symbol.Underlying.Value : 
```

Comment:
Currently derivatives are represented differently in Symbol class. e.g. CFDs don't have non-null underlying property. So checking for null underlying won't cover all derivatives. I am planning to harmonize it a bit later to have a uniform derivatives representation. As for behavior itself, I will add now options as well - very soon we will be supporting index options and market hours will be different.